plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -126,4 +239,99 @@ AS SELECT sum(value) AS s FROM iceberg2.tpch.common_base_table
         assertUpdate(defaultIceberg, "DROP TABLE common_base_table");
         assertUpdate("DROP MATERIALIZED VIEW mv_on_iceberg2");
     }
+
+    private int validateResultContains2AndReturnSecondValue(List<MaterializedRow> materializedRows)
+    {
+        assertThat(materializedRows.size()).isEqualTo(2);
+        assertThat(materializedRows.stream().filter(value -> (int) value.getField(0) == 2).count()).isEqualTo(1);
+        return materializedRows.stream().mapToInt(value -> (int) value.getField(0)).filter(value -> value != 2).findFirst().orElseThrow();
+    }
+
+    public static class RandomTableFunction
+            extends AbstractConnectorTableFunction
+    {
+        public RandomTableFunction()
+        {
+            super("system", "random_function", List.of(), GENERIC_TABLE);
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments, ConnectorAccessControl accessControl)
+        {
+            return TableFunctionAnalysis.builder()
+                    .handle(new RandomTableFunctionHandle())
+                    .returnedType(new Descriptor(ImmutableList.of(new Descriptor.Field("random_value", Optional.of(INTEGER)))))
+                    .build();
+        }
+    }
+
+    public  static class RandomTableFunctionHandle
+            implements ConnectorTableFunctionHandle
+    { }
+
+    public  static class RandomTableFunctionProcessorProvider
+            implements TableFunctionProcessorProvider
+    {
+        private final RandomFunctionProcessor randomFunctionProcessor = new RandomFunctionProcessor();
+        @Override
+        public TableFunctionSplitProcessor getSplitProcessor(ConnectorSession session, ConnectorTableFunctionHandle handle, ConnectorSplit split)
+        {
+            randomFunctionProcessor.reset();
+            return randomFunctionProcessor;
+        }
+    }
+
+    public static class RandomFunctionProcessor
+            implements TableFunctionSplitProcessor
+    {
+        // seed is provided just to make sure that returned numbers won't cause tests to fail
+        private final Random rand = new Random(13433);
```

Comment:
for the test it's important that RandomFunctionProcessor is re-used.
in a sense, this field could be static.

in fact, we don't need randomness at all. `static final AtomicInteger` would be even better

Reply:
yes, I will rework this